### PR TITLE
Collijk/feature/new grid plots

### PR DESCRIPTION
**Pull request by collijk**
Brand spankin' new grid plots!  
image

image

### CODE REVIEW

File: specification_templates/diagnostics.yaml [hunk 1]
```diff
@@ -17,36 +17,36 @@ grid_plots:
...
     comparators:
       # Full path or DATE.VV or latest/best
-      - version: '/ihme/covid-19/seir-outputs/2021_01_13.08'
+      - version: '/ihme/covid-19/seir-outputs/2021_01_18.05'
         scenarios:
           # Keys are scenarios and values are plot labels. Within a job,
           # good idea to keep the plot labels unique.
           worse: 'worse'
-          reference: 'reference'
+          reference*: 'reference'
```
Comment (miker985) [inline specification_templates/diagnostics.yaml:25]: an asterisks?

File: src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plot_version.py [hunk 2]
```diff
@@ -106,9 +106,14 @@ def clean_data(self, data: pd.DataFrame, location_id: Optional[int]):
...
 
 def make_plot_versions(comparators: List[GridPlotsComparatorSpecification], color_map) -> List[PlotVersion]:
+    primary_versions = []
     plot_versions = []
     for comparator in comparators:
         for scenario, label in comparator.scenarios.items():
-            plot_versions.append((Path(comparator.version), scenario, label))
-    plot_versions = [PlotVersion(*pv, color_map(i)) for i, pv in enumerate(plot_versions)]
-    return plot_versions
+            if scenario.endswith('*'):
```
Comment (miker985) [inline src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plot_version.py:113]: ok, I get the asterisks now. I'm cringing a little, but not too much...

File: src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plotter.py [hunk 3]
```diff
@@ -17,9 +17,13 @@
...
-COLOR_MAP = plt.get_cmap('Set1')
-DATE_LOCATOR = mdates.AutoDateLocator(maxticks=10)
-DATE_FORMATTER = mdates.ConciseDateFormatter(DATE_LOCATOR, show_offset=False)
+COLOR_MAP = lambda i: ['#7F3C8D', '#11A579',
+                       '#3969AC', '#F2B701',
+                       '#E73F74', '#80BA5A',
+                       '#E68310', '#008695',
+                       '#CF1C90', '#f97b72',
+                       '#4b4b8f', '#A5AA99'][i]
+
```
Comment (miker985) [inline src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plotter.py:26]: COLOR_MAP = [...].__getitem__ ?

File: src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plotter.py [hunk 4]
```diff
@@ -57,63 +60,122 @@ def make_covariates_page(plot_versions: List[PlotVersion],
...
+        start, end,
+        label='Log Beta Residuals',
+        vlines=vlines,
+    )
+    
+    ax_rhist = fig.add_subplot(gs_beta[2])
+    make_log_beta_resid_hist(
+        ax_rhist,
+        plot_versions,
+        location.id
```
Comment (miker985) [inline src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plotter.py:176]: trailing comma

File: src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plotter.py [hunk 5]
```diff
@@ -242,28 +323,101 @@ def make_results_page(plot_versions: List[PlotVersion],
...
+    
+    make_time_plot(
+        ax_susceptible,
+        plot_versions,
+        'total_susceptible',
+        location.id,
+        start, end,
+        label='Total Susceptible (% Population)',
+        vlines=vlines,
+        transform=lambda x: 100 * x / pop
```
Comment (miker985) [inline src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plotter.py:337]: trailing comma

**Review by miker985 — APPROVED** [review]

File: src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plot_version.py (hunk 2 above)
Comment (collijk) [reply src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plot_version.py:113]: Sorry not sorry.

File: src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plotter.py (hunk 3 above)
Comment (collijk) [reply src/covid_model_seiir_pipeline/pipeline/diagnostics/model/plotter.py:26]: I like it.

**Pull request merged**